system-tests/lib/cre/environment/jobs.go

```
@@ -56,13 +56,43 @@ func StartJD(lggr zerolog.Logger, nixShell *nix.Shell, jdInput jd.Input, infraIn
 	return jdOutput, nil
 }
 
-func SetupJobs(lggr zerolog.Logger, jdInput jd.Input, nixShell *nix.Shell, registryChainBlockchainOutput *blockchain.Output, topology *cre.Topology, infraInput infra.Input, capabilitiesAwareNodeSets []*cre.CapabilitiesAwareNodeSet) (*jd.Output, []*cre.WrappedNodeOutput, error) {
+func CreateJobDistributor(input *jd.Input) (*jd.Output, error) {
+	if os.Getenv("CI") == "true" {
+		jdImage := ctfconfig.MustReadEnvVar_String(E2eJobDistributorImageEnvVarName)
+		jdVersion := os.Getenv(E2eJobDistributorVersionEnvVarName)
+		input.Image = fmt.Sprintf("%s:%s", jdImage, jdVersion)
+	}
+
+	jdOutput, err := jd.NewJD(input)
+	if err != nil {
+		return nil, pkgerrors.Wrap(err, "failed to create new job distributor")
+	}
+
+	return jdOutput, nil
+}
+
+func StartDONsAndJD(lggr zerolog.Logger, jdInput *jd.Input, registryChainBlockchainOutput *blockchain.Output, topology *cre.Topology, infraInput *infra.Input, capabilitiesAwareNodeSets []*cre.CapabilitiesAwareNodeSet) (*jd.Output, []*cre.WrappedNodeOutput, error) {
+	if jdInput == nil {
+		return nil, nil, errors.New("jd input is nil")
+	}
+	if infraInput == nil {
+		return nil, nil, errors.New("infra input is nil")
+	}
+	if registryChainBlockchainOutput == nil {
+		return nil, nil, errors.New("registry chain blockchain output is nil")
+	}
+	if topology == nil {
+		return nil, nil, errors.New("topology is nil")
+	}
+	if len(capabilitiesAwareNodeSets) == 0 {
```

IMO you can skip a lot of this validation: just let the application panic which will provide a stacktrace that explains what's happening

Reasoning is that these APIs are internal to us